### ISSUE TITLE
Music overlaps randomly

### ISSUE THREAD

**Issue by ja2forever** [ja2-stracciatella/ja2-stracciatella#1786]
| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  ja2-stracciatella_0.20.0-git+439a6d3_win-msvc64 | JA2 Gold 1.12 English CD  |    no    |    ?    |

Music sometimes starts to overlap for no reason and randomly. Most noticable on themes without enemy fights.

**Comment by lynxlynxlynx** [comment]
What do you mean by overlap? That two songs are playing at the same time while they shouldn't? For how long, is it just a transitional problem?

Unfortunately without a reproducible case, we can't do much. I can't say I've experienced it either.

**Comment by ja2forever** [comment]
yes exactly like one music plays then suddenly second starts playing. I cant really tell if something is causing this. Should i screen record with sound and post here video?

**Comment by selaux** [comment]
Is it maybe trying to fade over between the two music pieces? How long does the overlap take?

**Comment by ja2forever** [comment]
Randomly pretty. Most noticabe on start of game while in first sector - usually happens but not always. Right now happend and only thing i did was: launch game - new game - hire aim x1 - close laptop - go to sector - alt + o to kill enemies and shortly after that like 1 minute or so this bug.

**Comment by ChessPhishGuy** [comment]
I read about this issue this morning and noticed that I have the same issue. Two songs playing at the same time. Might have started after I left and returned to app. Like it forgot it was playing one song and started playing another at the same time.

**Comment by ChessPhishGuy** [comment]
Started two audio tracks of the same song when entering laptop off by a fraction of a second to where I hear both like an echo.

**Comment by momoko-h** [comment]
Could be a race condition. SDL executes the audio callback function an separate thread, and we're not doing anything to protect against concurrent accesses. The mutex in Soundman only protects the thread that is filling the ring buffers.
